src/consensus/raft/raft.h

```diff
@@ -1170,11 +1170,7 @@ namespace raft
       LOG_DEBUG_FMT("Compacting...");
       if (state == Leader)
       {
-        auto snapshot_idx = snapshotter->snapshot(idx);
-        if (snapshot_idx.has_value())
-        {
-          last_snapshot_idx = snapshot_idx.value();
-        }
+        snapshotter->snapshot(idx);
```

Comment:
Raft doesn't need to know about the last snapshot idx. Instead, it will ask the snapshotter for the latest snapshot idx when a new node is added.